src/mappings/shared/token/setMetadata.ts

```diff
@@ -26,3 +26,35 @@ export async function setMetadataHandler(context: Context, collection: CE, nft:
   const existingToken = await getOptional<TE>(context.store, TE, generateTokenId(collection.id, nftMedia))
   return await (existingToken ? tokenAPI.addNftToToken(nft, existingToken) : tokenAPI.create(collection, nft))
 }
+
+export async function setMetadataOnCollectionHandler(context: Context, collection: CE): Promise<TE[] | undefined> {
```

Comment:
can't this be done in `setMetadataHandler`?

Reply:
i find it more clean when they are separated.
if you disagree, lmk and i'll merge them

Reply:
in that case I would change `setMetadataHandler` to be `setMetadataOnNftHandler` if it only handle nft metadata

Reply:
aight

Reply:
done